src/adapters/bidderFactory.js

```diff
@@ -0,0 +1,240 @@
+import Adapter from 'src/adapter';
+import bidmanager from 'src/bidmanager';
+import bidfactory from 'src/bidfactory';
+import { STATUS } from 'src/constants';
+
+import { ajax, GET, POST } from 'src/ajax';
+import { logWarn, logError, parseQueryStringParameters, delayExecution } from 'src/utils';
+
+/**
+ * This file aims to support Adapters during the Prebid 0.x -> 1.x transition.
+ *
+ * Prebid 1.x and Prebid 0.x will be in separate branches--perhaps for a long time.
+ * This function defines an API for adapter construction which is compatible with both versions.
+ * Adapters which use it can maintain their code in master, and only this file will need to change
+ * in the 1.x branch.
+ *
+ * Typical usage looks something like:
+ *
+ * const adapter = newBidder({
+ *   code: 'myBidderCode',
+ *   areParamsValid: function(paramsObject) { return true/false },
+ *   buildRequests: function(bidRequests) { return some ServerRequest(s) },
+ *   interpretResponse: function(oneServerResponse) { return some Bids, or throw an error. }
+ * });
+ */
+
+/**
+ * This function aims to minimize the Adapter-specific code for "single request" Bidders. That is, if your adapter
+ * only makes a single request to the server for bids, regardless of how many bid requests or ad units are in
+ * the Auction, then this function is for you.
+ *
+ * @param {BidderSpec} spec An object containing the bare-bones functions we need to make a Bidder.
+ */
+export default function newBidder(spec) {
+  return Object.assign(new Adapter(spec.code), {
+    callBids: function(bidsRequest) {
+      if (!bidsRequest.bids || !bidsRequest.bids.filter) {
+        return;
+      }
+
+      // callBids must add a NO_BID response for _every_ placementCode, in order for the auction to
+      // end properly. This map stores placement codes which we've made _real_ bids on.
+      //
+      // As we add _real_ bids to the bidmanager, we'll log the placementCodes here too. Once all the real
+      // bids have been added, fillNoBids() can be called to end the auction.
+      //
+      // In Prebid 1.0, this will be simplified to use the `addBidResponse` and `done` callbacks.
+      const placementCodesHandled = {};
+      function addBidWithCode(placementCode, bid) {
+        placementCodesHandled[placementCode] = true;
+        bidmanager.addBidResponse(placementCode, bid);
+      }
+      function fillNoBids() {
+        bidsRequest.bids
+          .map(bidRequest => bidRequest.placementCode)
+          .forEach(placementCode => {
+            if (placementCode && !placementCodesHandled[placementCode]) {
+              bidmanager.addBidResponse(placementCode, newEmptyBid());
+            }
+          });
+      }
+
+      const bidRequests = bidsRequest.bids.filter(filterAndWarn);
```

Comment:
These names are confusing, especially here.  I think `bidRequests` is fine, but `bidsRequest` I think is called `bidderRequest` in other parts of Prebid.